scripts/coverage_test

```diff
@@ -1,16 +1,25 @@
 #!/bin/sh
 set -e
 
-cc-test-reporter before-build
+# See the documentation for why we need this if statement
+# https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
+if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
```

Comment:
I could also break this into another script if you'd expect `coverage_test` to always produce coverage.

I could also write a function that wraps codeclimate behaviors that just doesn't report if it's a travis PR.

Open to ideas - this was just the easiest / first thing that came to mind.

Reply:
hm, this is probably OK because the `go test` line is different. The other option i'd consider is two different files but then the switch has to live outside of it... which is kinda weird too.

Reply:
The switch living outside is actually similar to what they recommend in the docs linked above:

```yaml
script:
   - 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then bash ./travis/run_on_pull_requests; fi'
   - 'if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then bash ./travis/run_on_non_pull_requests; fi'
```

Reply:
I actually think that's clearer - making the update.

Reply:
minus the != false. why, travis, why!?